### PR TITLE
advanced blocking response

### PR DESCRIPTION
**Pull request by noamlevipx**

### CODE REVIEW

File: src/main/java/com/perimeterx/api/blockhandler/DefaultBlockHandler.java [hunk 1]
```diff
@@ -67,11 +67,49 @@ private void sendMessage(String blockPageResponse, HttpServletResponseWrapper re
...
+                responseWrapper.setContentType(Constants.CONTENT_TYPE_TEXT_HTML);
+            }
         }
         responseWrapper.getWriter().print(blockPageResponse);
     }
 
+    private boolean shouldHandleAdvancedBlockingResponse(PXContext context) {
+
+        //if advanced blocking response config is disabled
+        if (!context.isAdvancedBlockingResponse())
```
Comment (yaronpx) [inline src/main/java/com/perimeterx/api/blockhandler/DefaultBlockHandler.java]: code style: always use curly braces on `if` statements

File: src/main/java/com/perimeterx/models/httpmodels/AdvancedBlockingResponse.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    private String uuid;
+
+    @JsonProperty("hostUrl")
+    private String hostUrl;
+
+    @JsonProperty("blockScript")
+    private String blockScript;
+
+
+    public AdvancedBlockingResponse(String appId, String jsClientSrc, String firstPartyEnabled, String vid, String uuid, String hostUrl, String blockScript)
```
Comment (yaronpx) [inline src/main/java/com/perimeterx/models/httpmodels/AdvancedBlockingResponse.java]: code style: opening braces at the end of line, not in a new line

File: src/main/java/com/perimeterx/api/blockhandler/DefaultBlockHandler.java [hunk 3]
```diff
@@ -67,11 +67,49 @@ private void sendMessage(String blockPageResponse, HttpServletResponseWrapper re
...
+        String header = context.getHeaders().get("accept");
+        if (header != null && header.toLowerCase().contains("application/json")) {
+            headerExists = true;
+        } else {
+            header = context.getHeaders().get("content-type");
+            headerExists = header != null && header.toLowerCase().contains("application/json");
+        }
+
+        return headerExists
+                && context.getCookieOrigin().equals(Constants.COOKIE_HEADER_NAME)
```
Comment (yaronpx) [inline src/main/java/com/perimeterx/api/blockhandler/DefaultBlockHandler.java]: any chance that `context.getCookieOrigin()` might be `null`?

Comment (noamlevipx) [reply src/main/java/com/perimeterx/api/blockhandler/DefaultBlockHandler.java]: no

**Pull request merged**